yaksh/views.py

```diff
@@ -960,3 +988,43 @@ def design_questionpaper(request):
         context = {'form': form, 'questionpaper':True}
         return my_render_to_response('yaksh/design_questionpaper.html',
                                      context, context_instance=ci)
+
+@login_required
+def view_profile(request):
+    """ view moderators and users profile """
+
+    user = request.user
+    return my_render_to_response('yaksh/view_profile.html', {'user':user})
+
+@login_required
+def edit_profile(request):
+    """ edit profile details facility for moderator and students """
+
+    context = {}
+    user = request.user
+    ci = RequestContext(request)
+    profile = Profile.objects.get(user_id=user.id)
+    if request.method == 'POST':
+        form = EditProfile(user, request.POST)
+        if form.is_valid():
+            user.first_name = request.POST['first_name']
+            user.last_name = request.POST['last_name']
+            profile.department = request.POST['department']
```

Comment:
You can simply do form_data = form.save(commit=False) here profile will be updated and do not forget to pass profile instance to the form.